modules/phase_field/src/auxkernels/FeatureFloodCountAux.C

```diff
@@ -25,23 +30,23 @@ FeatureFloodCountAux::FeatureFloodCountAux(const InputParameters & parameters) :
     _flood_counter(getUserObject<FeatureFloodCount>("bubble_object")),
     _var_idx(getParam<unsigned int>("map_index")),
     _field_display(getParam<MooseEnum>("field_display")),
-    _var_coloring(_field_display == "VARIABLE_COLORING")
+    _var_coloring(_field_display == "VARIABLE_COLORING"),
+    _field_type(static_cast<FeatureFloodCount::FIELD_TYPE>(static_cast<int>(_field_display)))
 {
   if (isNodal())
   {
-    if (_field_display == "CENTROID")
-      mooseError("CENTROID coloring is only available for elemental aux variables");
-
     if (_field_display == "ACTIVE_BOUNDS")
       mooseError("ACTIVE_BOUNDS is only available for elemental aux variables");
 
-    if (_flood_counter.isElemental() && (_field_display == "UNIQUE_REGION" || _field_display == "VARIABLE_COLORING"))
-      mooseError("UNIQUE_REGION and VARIABLE_COLORING must be on variable types that match the entity mode of the FeatureFloodCounter");
+    if (_flood_counter.isElemental() &&
+        (_field_display == "UNIQUE_REGION" || _field_display == "VARIABLE_COLORING" || _field_display == "GHOSTED_ENTITIES" || _field_display == "HALOS"))
+      mooseError("UNIQUE_REGION, VARIABLE_COLORING, and GHOSTED_ENTITITES must be on variable types that match the entity mode of the FeatureFloodCounter");
   }
   else
   {
-    if (! _flood_counter.isElemental() && (_field_display == "UNIQUE_REGION" || _field_display == "VARIABLE_COLORING"))
-      mooseError("UNIQUE_REGION and VARIABLE_COLORING must be on variable types that match the entity mode of the FeatureFloodCounter");
+    if (! _flood_counter.isElemental() &&
```

Comment:
this could be move out of the enclosing if/else with the condition changed to `_flood_counter.isElemental() != isNodal()` (then you can do with one copy of this test)